File: Algorithms/Medium/53_MaximumSubarray/Solution.java
class Solution {
    public int maxSubArray(int[] nums) {
        int maxSum = Integer.MIN_VALUE;
        int sum = 0;

        for (int num : nums) {
            sum += num;
            maxSum = Math.max(sum, maxSum);

            if (sum < 0) {
                sum = 0;
            }
        }

        return maxSum;
    }
}

File: Algorithms/Medium/53_MaximumSubarray/Solution.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        current_sum, max_sub_sum = 0, nums[0]

        for num in nums:
            if current_sum < 0:
                current_sum = 0
                
            current_sum += num 
            max_sub_sum = max(current_sum, max_sub_sum)

        return max_sub_sum